src/Repository/DefaultUnleashRepository.php

```diff
@@ -239,19 +247,161 @@ private function parseFeatures(array $body): array
             }
 
             $featureVariants = $this->parseVariants($feature['variants'] ?? []);
+            $dependencies = $this->parseDependencies($feature['dependencies'] ?? [], $features, $hasUnresolvedDependencies);
 
-            $features[$feature['name']] = new DefaultFeature(
+            $featureDto = new DefaultFeature(
                 $feature['name'],
                 $feature['enabled'],
                 $strategies,
                 $featureVariants,
                 $feature['impressionData'] ?? false,
+                $dependencies,
             );
+            if ($hasUnresolvedDependencies) {
+                $unresolved[] = $featureDto;
+            } else {
+                $features[$feature['name']] = $featureDto;
+            }
         }
 
+        // both parameters are passed by reference so that the method can modify them both
+        $this->handleUnresolvedDependencies($unresolved, $features);
+
         return $features;
     }
 
+    /**
+     * @param array<Feature>                $unresolved
+     * @param array<string, DefaultFeature> $features
+     */
+    private function handleUnresolvedDependencies(array &$unresolved, array &$features): void
```

Comment:
This part seems to be very different from all the other Unleash SDKs I've seen. Do we have any README/design decisions/ADRs in this repo that would explain why we're doing this part?

Reply:
Not really, I haven't looked at the other SDKs and this is just how I did this.